circuit_knitting_toolbox/circuit_cutting/cutting_evaluation.py

```diff
@@ -14,9 +14,112 @@
 from __future__ import annotations
 
 from collections.abc import Sequence
+from itertools import chain
+from multiprocessing.pool import ThreadPool
 
+import numpy as np
 from qiskit.circuit import QuantumCircuit, ClassicalRegister
-from ..utils.observable_grouping import CommutingObservableGroup
+from qiskit.quantum_info import PauliList
+from qiskit.providers import Backend
+from qiskit.primitives.base import BaseSampler
+from qiskit.result import QuasiDistribution
+from qiskit_aer import AerSimulator
+
+from ..utils.observable_grouping import CommutingObservableGroup, ObservableCollection
+from ..utils.iteration import strict_zip
+from ..utils.simulation import ExactSampler
+from .qpd import (
+    QPDBasis,
+    SingleQubitQPDGate,
+    TwoQubitQPDGate,
+    generate_qpd_samples,
+    decompose_qpd_instructions,
+)
+from .cutting_decomposition import decompose_observables
+
+
+def execute_experiments(
+    circuits: QuantumCircuit | dict[str | int, QuantumCircuit],
+    observables: PauliList | dict[str | int, PauliList],
+    num_samples: int,
+    samplers: BaseSampler | Sequence[BaseSampler] | None = None,
+) -> tuple[list[list[list[tuple[QuasiDistribution, int]]]], Sequence[float]]:
```

Comment:
It'd be nice to use `namedtuple` for these functions that return multiple things (i.e., this and `partition_problem`).  Not urgent, as this change will be backwards compatible.